Refactor box loading to be more explicit

ekohl opened the pull request:
The foreman repositories no longer has an EL6 repository so this can't
work anymore.

ekohl commented:
Updated the loader and tests but now I realise this also removes all the older versions. We need to rethink this because I'm not sure what's the best solution here.

ekohl commented:
@stbenjam or @ehelms any idea what would be a good way to keep centos 6 with older foreman releases but not nightly? I think we may need some sort of support matrix.

ehelms commented:
@ekohl I meant to indicate I had seen this and was still pondering on it :)

ekohl commented:
I rewrote it. Please have a look if the general idea is correct. I know this is currently broken and also removes the puppet 4 configs but it's mostly about the versions.yaml rewrite.

ekohl commented:
Oh, and it will most likely go very nicely together with https://github.com/Katello/forklift/pull/322 and works towards https://github.com/Katello/forklift/issues/302

ehelms commented:
I *think* with your re-write we could remove this convolutedness I wrote -- https://github.com/Katello/forklift/blob/master/playbooks/roles/katello_repositories/defaults/main.yml#L2-L7

ehelms commented:
With your re-factor this moves us strictly on to Vagrant 1.8+ due to the use of host_vars (which is great). So I think this PR is a good candidate to add a 'Requirements' section to the top end of the Readme to spell out our minimums.

ehelms commented:
Overall I like it, the code itself looks simplified and the config is quite obvious. Keeps the generation dynamic but since its more simplified I think it will be more obvious when debugging.

ekohl commented:
Ok, I'm testing around a bit and will come up with a rebased PR when #322 has been merged since that makes supporting foreman boxes much easier.

ekohl commented:
I still need to update the README, fix the tests, verify the capsule boxes work. I also started with using the atlas boxes because the centos 6 box couldn't connect to EPEL anymore but the atlas box is lacking libselinux-python which ansible doesn't like. Haven't tried to fix that yet.

ekohl commented:
Updated. I tested:
* Provision centos6-foreman-1.12
* Provision centos7-katello-nightly
* Provision centos7-capsule-nightly

I'll try to test other boxes for the atlas box update though I may use a similar approach as in https://github.com/Katello/forklift/pull/316 with legacy boxes.

ehelms commented on README.md:
```diff
@@ -18,7 +18,7 @@ Forklift provides tools to create Foreman/Katello environments for development,
 ### Quickstart
 
-This will walk through the simplest path of spinning up a production test environment of a bleeding edge nightly installation assuming vagrant and Libvirt are installed and configured.
+This will walk through the simplest path of spinning up a production test environment of a bleeding edge nightly installation assuming Vagrant (version 1.8+) and Libvirt are installed and configured.
```
What do you think about adding something like:

```
### Requirements

 * Vagrant - 1.8+
 * Ansible - 2.1+
```

ekohl replied:
I like it, will change this.

ehelms commented on playbooks/roles/selinux/tasks/main.yml:
```diff
@@ -0,0 +1,5 @@
+- name: 'Ensure libselinux-python is installed'
+  yum:
+    name: 'libselinux-python'
+    state: installed
```
This would be a good role to also had enforce/permissive setting of SELinux which is sometimes needed for testing or devel setups.

ehelms replied:
And could help ensure SELinux is enforcing by default in case the box someone uses has it disabled for any reason

ekohl replied:
I'll look into this.

ekohl commented:
Updated:
* Selinux is now set to enforcing but is a parameter (ready for the devel box)
* Documented the minimal version in README
* Fixed the Rubocop failures
* Fixed provisioning the freeipa server but I don't think it properly sets up the katello server

ehelms commented:
This inspired me to open up https://github.com/Katello/forklift/pull/326 for discussion

ekohl commented:
I think this is ready for merge though I'm still a bit unsure about the pipeline.

ehelms commented:
@ekohl Apologies, forgot about this one! Will prioritize re-review

ehelms commented:
Looks great - going to merge and see what happens :)

lpramuk commented:
Format change of ```config/versions.yaml``` has broken ```katello_version``` mapping func in ```forklift/lib/forklift/repositories.rb```:
```
    def katello_version
      @versions['mapping'][@version]
    end
```

ekohl commented:
Could you try replacing it with
```ruby
def katello_version
  @versions['installers'].select {|scenario| scenario['foreman'] == @version }['katello'] rescue nil
end
```

lpramuk commented:
Like almost, you missed ```[0]```
```
@versions['installers'].select {|scenario| scenario['foreman']=='nightly'}[0]['katello']
```
